### PR TITLE
Add oslogin terraform module for optionally configuring OSLogin per project.

### PR DESCRIPTION
**Pull request by var-sdk**
Work towards https://github.com/sigstore/public-good-instance/issues/376. OSLogin enables us to control access to VMs without requiring provisioning of ssh keys and has better audit properties. The module is off by default so this should be a no-op.

### CODE REVIEW

File: terraform/gcp/modules/oslogin/general.auto.tfvars [hunk 1]
```diff
@@ -0,0 +1,17 @@
+project_id = "projectsigstore-staging"
```
Comment (cpanato) [inline terraform/gcp/modules/oslogin/general.auto.tfvars:1]: Do we need this file?

Comment (var-sdk) [reply terraform/gcp/modules/oslogin/general.auto.tfvars]: Oops, no that was a test file. Thanks for catching. Removed.

File: terraform/gcp/modules/sigstore/sigstore.tf [hunk 2]
```diff
@@ -225,3 +225,29 @@ module "project_roles" {
...
   iam_members_to_roles = var.iam_members_to_roles
 }
+
+// OSLogin configuration
+module "oslogin" {
+  source     = "../oslogin"
+  project_id = var.project_id
+
+  // Disable module entirely if oslogin is disabled
+  count = var.monitoring.enabled ? 1 : 0
```
Comment (priyawadhwa) [inline terraform/gcp/modules/sigstore/sigstore.tf:235]: this should be `var.oslogin.enabled`

Comment (var-sdk) [reply terraform/gcp/modules/sigstore/sigstore.tf]: Fixed.

**Review by priyawadhwa — APPROVED** [review]
lgtm!

**Review by cpanato — APPROVED** [review]

**Comment by cpanato** [comment]
@priyawadhwa it is good to merge?

**Pull request merged**